src/store/mod.rs

```diff
@@ -0,0 +1,240 @@
+use crate::node_pages_map::{Page, PAGE_SIZE};
+use crossbeam_channel::{bounded, unbounded, Receiver, Sender};
+use io_uring::{cqueue, opcode, squeue, types, IoUring};
+use std::{fs::File, os::fd::RawFd};
+use std::{ops::Range, os::fd::AsRawFd, path::PathBuf, sync::Arc};
+
+mod completion;
+#[cfg(test)]
+mod tests;
+
+/// The Store is an on disk array of [`crate::node_pages_map::Page`]
+// TODO: make the store thread-safe so that it can be accessed from multiple threads
+pub struct Store {
+    store_file: File,
+    capacity: u64,
+    ring: Arc<IoUring>,
+    ring_capacity: u32,
+    tx_new_batches: Sender<(Range<u64>, Option<Vec<Box<Page>>>, Sender<()>)>,
+    tx_kill_completion_handler: Sender<()>,
+    nonce: u64,
+}
+
+/// Options to open a Store
+pub struct StoreOptions {
+    /// File path of the storage file.
+    file_path: PathBuf,
+    /// Max number of pages supported
+    pages_capacity: u64,
+    /// Maximum number of entries supported in the io_uring submission queue
+    io_uring_capacity: u32,
+    /// Reset the store to an empty state
+    reset: bool,
+}
+
+impl Default for StoreOptions {
+    fn default() -> Self {
+        Self {
+            file_path: PathBuf::from("node_pages_store"),
+            pages_capacity: 1_000,
+            io_uring_capacity: 512,
+            reset: true,
+        }
+    }
+}
+
+impl Store {
+    /// Create a new Store given the StoreOptions
+    pub fn new(options: StoreOptions) -> Self {
+        if options.reset {
+            // TODO: Is there a better alternative? fallocate, trucate or head maybe
```

Comment:
well yes there is, first of all you can just zero it yourself. Then fallocate(FALLOC_FL_ZERO_RANGE) should do the job equally.

As we discussed, we want to allocate the space upfront, to reduce fragmentation and the number of extents, so the latter option is what we want to do eventually.